src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -1547,7 +1547,12 @@ void BoxLockNode::emit(C2_MacroAssembler* masm, PhaseRegAlloc* ra_) const
 uint BoxLockNode::size(PhaseRegAlloc *ra_) const
 {
   int offset = ra_->reg2offset(in_RegMask(0).find_first_elem());
-  return (offset < 0x80) ? 5 : 8; // REX
+  int reg = ra_->get_encode(this);
+  if (reg > 15) {
+    return (offset < 0x80) ? 6 : 9; // REX2
+  } else {
+    return (offset < 0x80) ? 5 : 8; // REX
+  }
```

Comment:
Please move this out to a separate patch.

Reply:
Could you please create the PR as you identified and fixed this issue? I will remove this code block once your updated BoxLock node fix is integrated into the mainline.